src/Storages/ObjectStorage/DataLakes/Iceberg/IcebergMetadataFilesCache.h

```diff
@@ -0,0 +1,187 @@
+#pragma once
+#include "config.h"
+
+#if USE_AVRO
+
+#include <Core/Settings.h>
+#include <Common/CacheBase.h>
+#include <Common/HashTable/Hash.h>
+#include <Common/logger_useful.h>
+#include <Storages/ObjectStorage/DataLakes/Iceberg/Snapshot.h>
+#include <Storages/ObjectStorage/DataLakes/Iceberg/ManifestFile.h>
+#include <Storages/ObjectStorage/StorageObjectStorage.h>
+
+namespace ProfileEvents
+{
+    extern const Event IcebergMetadataFilesCacheMisses;
+    extern const Event IcebergMetadataFilesCacheHits;
+    extern const Event IcebergMetadataFilesCacheWeightLost;
+}
+
+namespace CurrentMetrics
+{
+    extern const Metric IcebergMetadataFilesCacheSize;
+}
+
+namespace DB
+{
+
+/// Manifest list contains only a list of manifest files, so we don't have to store the deserialized manifest_list content.
+/// In ManifestListCacheCell we store a list of manifest files' data path and sequence number,
+/// so we can get manifest files from cache or read and construct them.
+struct ManifestListCacheCell
+{
+    struct ManifestFileEntry
+    {
+        String data_path;
```

Comment:
Did you mean `data_path` -> `manifest_file_path`? Naming it `data_path` can be confusing as it usually names actual data contents, not metadata.